iroha/src/block_sync.rs

```diff
@@ -75,10 +127,9 @@ impl BlockSynchronizer {
                     drop(
                         futures::future::join_all(
                             sumeragi
-                                .read()
+                                .send(GetSortedPeers)
                                 .await
-                                .network_topology
-                                .sorted_peers()
+                                .unwrap()
```

Comment:
I expect that we will not use unwraps when it will be a full pr not a draft, right?

Reply:
Idk actually. Generally if we recieve error here then we have some actor stopped, which means that actor paniced.
Generally that means that actor failed to initialized.

I would propose some type of safe unwrapping of this addresses. Something like this:
```rust
// iroha_actor/src/lib.rs

/// Actor for which safe to unwrap address, as we have some guarantees.
pub struct Infallible<A: InfallibleActor>(A);
```

Or just panic on sending messages to actors.

Reply:
It would be great to have Addr and InitializedAddr and call Addr.wait_initialized() for example. But this might be out of scope of this PR.

For this PR you can simply call `expect`

Reply:
Im saying about other type of initialization actually. This one requires some messages from other actors, like messages from kura with all the blocks.

Reply:
Maybe something like `sumeragi.expect_running().send()` or as this is our usual usecase we can just expect inside of `send` and add `send_handle_not_started` which let's us handle result

Reply:
`sumeragi.handle_not_running().send()`

Reply:
I bet we can create some function on addr like:
```
pub unsafe fn expect_running(self) -> AlwaysAddr<A> { ... }

impl AlwaysAddr<A> {
    fn send<M>(&self, m: M) -> M::Result { self.addr.send(m).unwrap() }
}
```

Reply:
I mean as sending with this `expect_running` is our usual case as it seems. It is better to structure it in other way round: by default `expect()` inside send, but let user `handle_not_running` and then he can get `Err` from send.

Reply:
I would prefer to have a type state btw :)